S2VX.Game/Story/Note/HoldNote.cs

```diff
@@ -8,5 +14,104 @@ public abstract class HoldNote : S2VXNote {
 
         [Resolved]
         private S2VXStory Story { get; set; }
+
+        [Resolved]
+        private ScreenStack Screens { get; set; }
+
+        [BackgroundDependencyLoader]
+        private void Load() => AddInternal(SliderPath);
+
+        private Path SliderPath { get; set; } = new Path() {
+            // Show slider path behind note
+            Depth = 100,
```

Comment:
What is this Depth and magic number?

Reply:
Using ChangeInternalChildDepth may be safer than setting it directly (there's some checking in there to make sure things are loaded and such).
![image](https://user-images.githubusercontent.com/12549910/101718157-8270e500-3a55-11eb-9eb1-896567c74cb4.png)

Reply:
I think ChangeInternalChildDepth or ChangeChildDepth is only needed if the following are true:
* You're trying to manipulate a child's Depth with the parent CompositeDrawable or Container object
* You're trying to manipulate the child's Depth after loading

I don't see anything in the logic of ChangeInternalChildDepth that does anything safer as you suggest:

```csharp
            var index = IndexOfInternal(child);
            if (index < 0)
                throw new InvalidOperationException($"Can not change depth of drawable which is not contained within this {nameof(CompositeDrawable)}.");

            internalChildren.RemoveAt(index);
            var aliveIndex = aliveInternalChildren.IndexOf(child);
            if (aliveIndex >= 0) // remove if found
                aliveInternalChildren.RemoveAt(aliveIndex);

            var chId = child.ChildID;
            child.ChildID = 0; // ensure Depth-change does not throw an exception
            child.Depth = newDepth;
            child.ChildID = chId;

            internalChildren.Add(child);
            if (aliveIndex >= 0) // re-add if it used to be in aliveInternalChildren
                aliveInternalChildren.Add(child);
```

It looks like it's just removing a child, setting the child's Depth, then re-adding the child back in. This makes sense when you're changing a child's depth after loading, but we don't need to do that if we're setting a child on initial load. This looks like a waste of work if I were to change HoldNote.cs:

```csharp
        [BackgroundDependencyLoader]
        private void Load() {
            AddInternal(SliderPath);
            ChangeInternalChildDepth(SliderDepth, 100);
        }
```

This would add SliderPath at Depth 0, remove SliderPath, set SliderPath's Depth to 100, and re-add SliderPath into the CompositeDrawable.

Reply:
I think ChangeInternalChildDepth or ChangeChildDepth is only needed if the following are true:
* You're trying to manipulate a child's Depth with the parent CompositeDrawable or Container object
* You're trying to manipulate the child's Depth after loading

Here's the bulk of the ChangeInternalChildDepth logic:

```csharp
            var index = IndexOfInternal(child);
            if (index < 0)
                throw new InvalidOperationException($"Can not change depth of drawable which is not contained within this {nameof(CompositeDrawable)}.");

            internalChildren.RemoveAt(index);
            var aliveIndex = aliveInternalChildren.IndexOf(child);
            if (aliveIndex >= 0) // remove if found
                aliveInternalChildren.RemoveAt(aliveIndex);

            var chId = child.ChildID;
            child.ChildID = 0; // ensure Depth-change does not throw an exception
            child.Depth = newDepth;
            child.ChildID = chId;

            internalChildren.Add(child);
            if (aliveIndex >= 0) // re-add if it used to be in aliveInternalChildren
                aliveInternalChildren.Add(child);
```

It looks like it removes a child, sets the child's Depth, then re-adds the child back in. This makes sense for safely changing a child's depth after loading, but we don't need to do that if we're setting a child on initial load. This looks like a waste of work if I were to change HoldNote.cs:

```csharp
        [BackgroundDependencyLoader]
        private void Load() {
            AddInternal(SliderPath);
            ChangeInternalChildDepth(SliderDepth, 100);
        }
```

This would add SliderPath with Depth 0, remove SliderPath from the CompositeDrawable, set SliderPath's Depth to 100, and re-add SliderPath back in to the CompositeDrawable.